tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -411,7 +411,7 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
 
             DataPipelineEnabled = config
                                   .WithKeys(ConfigurationKeys.TraceDataPipelineEnabled)
-                                  .AsBool(defaultValue: false);
+                                  .AsBool(defaultValue: EnvironmentHelpers.IsAzureAppServices() && FrameworkDescription.Instance.IsWindows());
```

Comment:
I think we actually want to check if the extension is installed to scope this as previously discussed
```suggestion
                                  .AsBool(defaultValue: EnvironmentHelpers.IsUsingAzureAppServicesSiteExtension() && FrameworkDescription.Instance.IsWindows());
```

Reply:
Good catch. This would have been enabled in Azure Functions using the nuget package, which doesn't include `libdatadog`.